guides/examples/elixir/Publisher.md

```diff
@@ -0,0 +1,111 @@
+# Publisher Example
+
+Ensure `:brod` is added to your deps on `mix.exs`
+
+```elixir
+defp deps do
+    [
+      {:brod, "~> 3.10.0"}
+    ]
+end
+```
+
+## Client Configuration
+
+Adding the following configuration (e.g. `config/dev.exs`)  
+
+```elixir
+import Config
+
+config :brod,
+  clients: [
+    kafka_client: [
+      endpoints: [localhost: 9092],
+      auto_start_producers: true,
+      # The following :ssl and :sasl configs are not 
+      # required when running kafka locally unauthenticated
+      ssl: true,
+      sasl: {
+        :plain,
+        System.get_env("KAFKA_CLUSTER_API_KEY"),
+        System.get_env("KAFKA_CLUSTER_API_SECRET")
+      }
+    ]
+  ]
+```
+
+_Note:_ `kafka_client` can be any valid atom. And `:endpoints` accepts multiple host port tuples (e.g. `endpoints: [{"192.168.0.2", 9092}, {"192.168.0.3", 9092}, ...]`).   
+
+## Publisher
+
+To send a message with brod we can use the `:brod.produce_sync/5` function
+
+```elixir
+defmodule BrodExample.Publisher do
+  def publish(topic, partition, key, message) do
+    :brod.produce_sync(:kafka_client, topic, partition, key, message)
+  end
+end
+```
+
+
+Now let's run and give it a try
+```
+➜  brod_sample git:(master) ✗ iex -S mix
+Erlang/OTP 22 [erts-10.7.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]
+
+
+10:58:41.442 [info]  [supervisor: {:local, :brod_sup}, started: [pid: #PID<0.210.0>, id: :kafka_client, mfargs: {:brod_client, :start_link, [[localhost: 9092], :kafka_client, [endpoints: [localhost: 9092], auto_start_producers: true]]}, restart_type: {:permanent, 10}, shutdown: 5000, child_type: :worker]]
+Interactive Elixir (1.10.2) - press Ctrl+C to exit (type h() ENTER for help)
+iex(1)>
+```
+
+Now, we can use our module, doing
+`iex(1)> BrodSample.Publisher.publish("streaming.events", 0, "", "Hello brod!")`
+If everything worked, brod will return a `:ok`
+
+So this sent the message `"Hello brod!"` to the topic named `"streaming.events"` on the partition number 0 and an empty partition key
+
+Let's take a look at kafdrop
+
+![kafdrop](../../../contrib/examples/elixir/docs/kafdrop.png)
+
+We can see that there is something on partition 0
+
+Opening it up we see
+![topic](../../../contrib/examples/elixir/docs/streamingevents.png)
+
+### Using partition key
+
+TODO: Update to talk about `:hash` as partitioner
+
+The most common way to send messages to kafka is by using a partition key and based on that deciding to what partition the message should go, let's see how we can achieve that
+
+First, we need to know how many partitions our topic have, so we don't try sending the message to a non-existing partition, for that we can also use brod
```

Comment:
We should perhaps give the readers a note that this is how brod maps key to partition number internally. Not something for one using brod to implement?

Reply:
I totally agree. We should just call out that `brod:produce_sync/3` does this. And _maybe_ give an example of a custom partition routing strategy.